src/renderers/dom/shared/HTMLDOMPropertyConfig.js

```diff
@@ -39,7 +39,7 @@ if (ExecutionEnvironment.canUseDOM) {
 
 var HTMLDOMPropertyConfig = {
   isCustomAttribute: RegExp.prototype.test.bind(
-    /^(data|aria)-[a-z_][a-z\d_.\-]*$/
+    new RegExp('^(data|aria)-[' + DOMProperty.ATTRIBUTE_NAME_CHAR + ']*$')
```

Comment:
Maybe it's a good idea to put all these regexes in one place like DOMProperty and just use constants.